### DISCUSSION TITLE
[General question] RDMA support

### DISCUSSION THREAD

**Discussion by wtao0221** [corundum/corundum#56] [General]
Hi,

I am just wondering whether the RDMA support in on the TODO list for Corundum. Which part can be modified to support RDMA?

**Comment by alexforencich** [comment]
Converting this to a discussion....

**Comment by alexforencich** [comment]
Right now RDMA support is basically a "wish list" feature.  I don't have time to work on it myself as there are a number of higher-priority items that are on the critical path for our experiments.  However, if there are enough folks in the community that are interested in not only using RDMA on corundum but also helping to implement RDMA on corundum, then I would be willing to support such an effort.  I have definitely had a few people ask about RDMA support, but so far nobody has expressed much interest in helping to implement such a feature.  

Unfortunately, another complicating factor is that I'm not very familiar with what is required for proper RDMA support on both the hardware side and the software side, so before diving into implementation, we'll have to get a handle on the high-level requirements and figure out how to reconcile that with the rest of corundum.  It's possible there will need to be some significant architectural changes required to support protocols like RDMA - after all, it's not a simple packet-processing block that can just be dropped in, it requires significant software support, possibly changes to the host interface, and it also turns requests back around on the NIC, which is not something that corundum is really set up to deal with at the moment.  

If you're interested in working on this (or if you know anyone who might be, or if anyone reading this might be), then please let me know and perhaps we can find a way to make that happen.

**Reply by liaoyunkun** [reply]
I have the experience on hardware and software with RDMA NIC, may be we can discuss. :-)

**Reply by alexforencich** [reply]
Sure, hop on the slack workspace!  Getting RDMA operational on Corundum is definitely something I want to do in the future; if you're interested in helping with the implementation, that would be great.

**Reply by liaoyunkun** [reply]
yeah, open-source RNIC is interesting for future research on In-Network computing or disaggregated data center.
